routers/routes/routes.go

```diff
@@ -149,22 +152,118 @@ func NewMacaron() *macaron.Macaron {
 			ExpiresAfter: setting.StaticCacheTime,
 		},
 	))
-	m.Use(public.StaticHandler(
-		setting.AvatarUploadPath,
-		&public.Options{
-			Prefix:       "avatars",
-			SkipLogging:  setting.DisableRouterLog,
-			ExpiresAfter: setting.StaticCacheTime,
-		},
-	))
-	m.Use(public.StaticHandler(
-		setting.RepositoryAvatarUploadPath,
-		&public.Options{
-			Prefix:       "repo-avatars",
-			SkipLogging:  setting.DisableRouterLog,
-			ExpiresAfter: setting.StaticCacheTime,
-		},
-	))
+
+	switch setting.Avatar.Storage.Type {
+	case "local":
+		m.Use(public.StaticHandler(
+			setting.Avatar.Path,
+			&public.Options{
+				Prefix:       "avatars",
+				SkipLogging:  setting.DisableRouterLog,
+				ExpiresAfter: setting.StaticCacheTime,
+			},
+		))
+	case "minio":
```

Comment:
Can't we just make the Storage a http.FileSystem?

Reply:
I rewrite this part but also not follow `http.FileSystem`, please review again.